File: PolyScript/frameworks/haskell/TestCompiler.hs
{-# LANGUAGE OverloadedStrings #-}
{-
Test compiler for Haskell PolyScript framework
Tests CRUD operations across all modes

Author: Mathew Burkitt, Dimension Technologies <mathew.burkitt@ditech.ai>
-}

module Main where

import PolyScript.Framework
import Data.Aeson (object, (.=))
import Data.Text (Text)
import qualified Data.Text as T
import Data.Time.Clock (getCurrentTime)
import System.Environment (getArgs)
import System.Exit (exitSuccess, exitFailure)

-- Test compiler tool implementation
data TestCompiler = TestCompiler

instance PolyScriptTool TestCompiler where
    toolDescription _ = "Test compiler for PolyScript framework"
    
    toolCreate _ resource _ _ = do
        let filename = maybe "test.txt" T.unpack resource
        return $ Right $ object
            [ "action" .= ("compile" :: Text)
            , "file" .= filename
            , "status" .= ("created" :: Text)
            ]
    
    toolRead _ Nothing _ _ = do
        -- List files
        return $ Right $ object
            [ "files" .= (["main.cpp", "utils.cpp", "test.cpp"] :: [Text])
            , "count" .= (3 :: Int)
            ]
    
    toolRead _ (Just resource) _ _ = do
        -- Read specific file
        return $ Right $ object
            [ "file" .= resource
            , "content" .= ("// Sample C++ code" :: Text)
            , "lines" .= (42 :: Int)
            ]
    
    toolUpdate _ resource _ _ = do
        let filename = maybe "test.txt" T.unpack resource
        return $ Right $ object
            [ "action" .= ("recompile" :: Text)
            , "file" .= filename
            , "status" .= ("updated" :: Text)
            ]
    
    toolDelete _ resource _ _ = do
        let filename = maybe "test.txt" T.unpack resource
        return $ Right $ object
            [ "action" .= ("clean" :: Text)
            , "file" .= filename
            , "status" .= ("deleted" :: Text)
            ]

main :: IO ()
main = runTool TestCompiler

File: PolyScript/frameworks/julia/test_compiler.jl
#!/usr/bin/env julia

#=
Test compiler for Julia PolyScript framework
Tests CRUD operations across all modes

Author: Mathew Burkitt, Dimension Technologies <mathew.burkitt@ditech.ai>
=#

push!(LOAD_PATH, @__DIR__)

using PolyScriptFramework

# Test compiler tool
struct TestCompiler <: PolyScriptTool end

# Implementation overrides
PolyScriptFramework.description(::TestCompiler) = "Test compiler for PolyScript framework"

function PolyScriptFramework.do_create(::TestCompiler, resource, options, ctx)
    filename = something(resource, "test.txt")
    Dict(
        "action" => "compile",
        "file" => filename,
        "status" => "created"
    )
end

function PolyScriptFramework.do_read(::TestCompiler, resource, options, ctx)
    if resource === nothing
        # List files
        Dict(
            "files" => ["main.cpp", "utils.cpp", "test.cpp"],
            "count" => 3
        )
    else
        # Read specific file
        Dict(
            "file" => resource,
            "content" => "// Sample C++ code",
            "lines" => 42
        )
    end
end

function PolyScriptFramework.do_update(::TestCompiler, resource, options, ctx)
    filename = something(resource, "test.txt")
    Dict(
        "action" => "recompile",
        "file" => filename,
        "status" => "updated"
    )
end

function PolyScriptFramework.do_delete(::TestCompiler, resource, options, ctx)
    filename = something(resource, "test.txt")
    Dict(
        "action" => "clean",
        "file" => filename,
        "status" => "deleted"
    )
end

# Run the tool
if abspath(PROGRAM_FILE) == @__FILE__
    exit(PolyScriptFramework.run_tool(TestCompiler()))
end